src/v/cluster/tm_stm_cache.cc

```diff
@@ -130,7 +130,17 @@ void tm_stm_cache::set_log(tm_transaction tx) {
       tx.etag,
       tx.pid,
       tx.tx_seq);
-    _log_txes[tx.id] = tx;
+
+    auto [tx_it, inserted] = _log_txes.try_emplace(tx.id, tx);
+    if (!inserted) {
+        tx_it->second.tx = tx;
+    }
```

Comment:
isn't this `.insert_or_assign(tx.id, tx)`?

Reply:
nope, "insert or update a field" not "insert or replace the object"

Reply:
oh yeh, now i see the `.tx`!